src/wallet/walletdb.h

```diff
@@ -274,6 +275,21 @@ class WalletBatch
     bool WriteActiveScriptPubKeyMan(uint8_t type, const uint256& id, bool internal);
     bool EraseActiveScriptPubKeyMan(uint8_t type, bool internal);
 
+    /**
+     * The active hd key for a wallet is the xpub of the xprv used to generate the
+     * automatically generated descriptors. However not all descriptor wallets will
+     * have this record, and so will need to be upgraded to include it.
+     * Additionally, it is possible to load a wallet that has had the upgrade performed
+     * into an older version of the software which does not know about this record,
+     * then generate new descriptors by encrypting, and then loading in a version that
+     * does know about the record. There would then be a mismatch between the stored
+     * xpub and the actual keys used to generate those descriptors. To account for this,
+     * we store whether the wallet was encrypted at the time this record was written,
+     * and can perform the upgrade again if the stored encryption status doesn't
+     * match the real status.
```

Comment:
Added a comment. The purpose is to be consistent so that the user is not confused. Otherwise an import could change the result of `gethdkey` unexpectedly. With the future `sethdseed` change, `gethdkey`'s result being different will at least be explicit, whereas an import may or may not change the detection of the hd key, depending on what else is/has been imported.